app/views/warehouse_reports/client_details/actives/index.xlsx.axlsx

```diff
@@ -15,29 +16,30 @@ wb.add_worksheet(name: "Active #{GrdaWarehouse::WarehouseReports::Dashboard::Bas
     'Living Situation',
     'Destination',
   ] + GrdaWarehouse::Hud::Client.race_fields.map { |m| HudUtility2024.race(m).gsub('None', 'Race None') }, style: title)
-  @report.enrollments.each do |client_id, enrollments|
-    enrollments.each do |enrollment|
-      client = enrollment.client
-      length_of_stay = if enrollment.last_date_in_program
-        enrollment.last_date_in_program - enrollment.first_date_in_program
-      else
-        Date.current - enrollment.first_date_in_program
-      end
 
-      row = [
-        client_id,
-        ::GrdaWarehouse::Config.get(:include_pii_in_detail_downloads) ? client.FirstName : 'Redacted',
-        ::GrdaWarehouse::Config.get(:include_pii_in_detail_downloads) ? client.LastName : 'Redacted',
-        HudUtility2024.project_type(enrollment.project_type),
-        enrollment.project.name(current_user),
-        enrollment.first_date_in_program,
-        enrollment.last_date_in_program,
-        length_of_stay.to_i,
-        enrollment.enrollment_group_id,
-        HudUtility2024.living_situation(enrollment.enrollment&.LivingSituation),
-        HudUtility2024.destination(enrollment.destination),
-      ] + GrdaWarehouse::Hud::Client.race_fields.map { |m| HudUtility2024.no_yes_reasons_for_missing_data(client.public_send(m.to_sym)&.to_i) }
-      sheet.add_row(row)
+  @report.enrollment_scope.preload(:client, :project, :enrollment).find_each do |enrollment|
+    client = enrollment.client
+    pii = client.pii_provider(user: current_user, context: :download)
```

Comment:
this introduces a lot of n+1 queries. Perhaps we should use ProjectPiiPolicy ?

Reply:
That works for me.  I wonder if we should plan to use ProjectPiiPolicy for reports for consistency.

Reply:
yeah, we could do that- I also wonder if we should plan implement bulk-preload for SouceClientPolicy and then replace ProjectPiiPolicy. If that seems reasonable I might leave this as-is for now. But if that seems unrealistic then I'll switch this to ProjectPiiPolicy, the only issue is that it doesn't handle all the other auth logic (roi, etc)

Reply:
per team check in about reporting perms, the ProjectPiiPolicy sounds like the best option